### DISCUSSION TITLE
Do the CT clamps fit in between DIN rail mounted modules?

### DISCUSSION THREAD

**Discussion by DavidS** [emporia-vue-local/esphome#303] [Q&A] [answered]
Hi,

thanks to the great work here, I'm considering getting a few Vue units for me and my parents. Looking at the specs of the clamps though, the seem to be pretty beefy, and I'm wondering whether I'll be able to fit them into my pretty crammed DIN rail based panel where all conductors (including the four conductors for incoming 3-phase) are spaced 18mm.

Anyone can confirm or deny that that's gonna work?

![image](https://github.com/user-attachments/assets/331bcb53-c00a-4f5b-9621-87b86aa04820)

**Comment by flaviut** [comment]
There's a dimensioned drawing of the clamps [in the official specs](https://www.emporiaenergy.com/wp-content/uploads/2024/03/Emporia-Vue-Gen-3-Technical-Specs.pdf), maybe that'll help you determine if it will fit.

There's also an [extensive discussion on third-party clamps](https://github.com/emporia-vue-local/esphome/discussions/59). TL;DR; pretty much any 0.333V current sensor will work just fine as well.

**Reply by DavidS** [reply]
Yeah, I saw that, and none of the dimensions on there are smaller than 18mm, hence why I'm asking if anyone had luck cramming them in despite this.

Since you mention replacement clamps: does every vue device come with ports for all 16 clamps, and I can just add whatever I want? 👀  that's neat.

**Reply by flaviut** [reply]
> if anyone had luck cramming them in despite this.

I see, sorry, I can't help here.

> does every vue device come with ports for all 16 clamps, and I can just add whatever I want?

Yup, exactly!

**Comment by MichaelWoodc** [comment]
I can't confirm or deny, but, it doesn't seem likely they'll fit in the current setup right off all the breakers.  You may be able to use half right off the breakers, and then put the rest a little further off on the wire, kindof offset or staggered if that makes sense.

**Comment by DavidS** [comment, marked as answer]
After much struggle, I can confirm that the CT-clamps that Emporia ships do not fit into European din-rail style breaker boxes. I've found [the clamps from openenergymonitor](https://shop.openenergymonitor.com/ct-sensors/) a much better fit. Below pics from the installation. The black ones are the ECS12-100, the white ECS06-20. As sold by openenergymonitor they come with 3.5mm jacks, so I had to also get some adapters to 2.5mm for the smaller ones.

![20241020_135058](https://github.com/user-attachments/assets/493024ff-7810-4e62-8a1d-ab44f2c99974)
![20241020_135054](https://github.com/user-attachments/assets/098bc742-e069-4b18-85d4-5cf4ce5927da)

PS: note that openenergymonitor is located in the UK, so on top of their hefty markup delivery to the EU also incurs additional import taxes. I'm now considering ordering two full sets of sensors directly from [echun](https://www.echun-elc.com/Products.asp?BigClassID=81&SmallClassID=153), as I was unable to find someone who was stocking those in the EU. Also there's a chance that ordering directly from them will enable getting them customized to the specs the Vue2 needs (2.5mm jacks, 0.333V).

PPS: this is a rental property and I am in no way responsible for these "Netzabkopplers".

**Reply by royduin** [reply]
Awesome! So those custom clamps work with the Emporia Vue 3? I also came across these super small clamps: https://www.brultech.com/store/index.php?id_product=45&controller=product, but not sure if they'll work with the Emporia Vue 3. Did you eventually import directly from echun? Any details?

**Reply by flaviut** [reply]
> Max continuous primary current Output: 26.23mA @ 40A

So they will not work as-is, but it should be pretty easy to add a burden resistor to bring it to ±.333V which is the emporia's signal type.

It's easier to do on the Vue 3 because the connectiors are better suited, this would be difficult on the Vue 2.

**Reply by royduin** [reply]
Good to know! I'll dig into it, thank you!
